### PR TITLE
Enable setting debug values via env vars

### PR DESCRIPTION
**Pull request by PatKamin**
Parse 4 new environment variables:
- SAMPLE_FREQUENCY
- PEBS_FREQ_HZ
- HOTNESS_MEASURE_WINDOW
- OLD_TIME_WINDOW_HOTNESS_WEIGHT

### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [ ] Other

### Checklist

- [ ] Code compiles without errors
- [ ] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [ ] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/3)

### CODE REVIEW

File: include/memkind/internal/memkind_memtier.h [hunk 1]
```diff
@@ -232,11 +232,8 @@ int memtier_ctl_set(struct memtier_builder *builder, const char *name,
 // float get_obj_hotness(int size);
 
 // PEBS
-// smaller value -> more frequent sampling
-// 10000 = around 100 samples on *my machine* / sec in matmul test
-#define SAMPLE_FREQUENCY 10000
-// #define PEBS_FREQ_HZ     20.0
-#define PEBS_FREQ_HZ     5.0
+extern unsigned long sample_frequency;
+extern float pebs_freq_hz;
```
Comment (maciekpac) [inline include/memkind/internal/memkind_memtier.h:236]: Why float? Unless vectorized or memory-bound, double operations should be as fast as float operations, with minimal memory overhead.

File: include/memkind/internal/memkind_memtier.h [hunk 2]
```diff
@@ -254,11 +251,8 @@ int memtier_ctl_set(struct memtier_builder *builder, const char *name,
 #define FINALIZE_HASH 0
 
 // hotness calculation
-#define HOTNESS_MEASURE_WINDOW 1000000000ULL
-// time window is 1s
-// #define OLD_TIME_WINDOW_HOTNESS_WEIGHT 0.999
-// #define OLD_TIME_WINDOW_HOTNESS_WEIGHT 0.4 // should not stay like this... only for tests and POC
-#define OLD_TIME_WINDOW_HOTNESS_WEIGHT  0.9 // should not stay like this... only for tests and POC
+extern unsigned long long hotness_measure_window;
+extern float old_time_window_hotness_weight;
```
Comment (maciekpac) [inline include/memkind/internal/memkind_memtier.h:255]: ditto - float vs double

File: src/memkind_memtier.c [hunk 3]
```diff
@@ -98,6 +98,12 @@
 #define THRESHOLD_CHECK_CNT 20
 #define THRESHOLD_STEP      1024
 
+//PEBS
+    float old_time_window_hotness_weight;
+    float pebs_freq_hz;
```
Comment (maciekpac) [inline src/memkind_memtier.c:103]: ditto: float vs double

File: src/memkind_memtier.c [hunk 4]
```diff
@@ -786,8 +792,38 @@ static struct memtier_memory *
...
+    if (env_var) {
+        hotness_measure_window = strtoull(env_var, NULL, 10);
+    }
+    env_var = memkind_get_env("SAMPLE_FREQUENCY");
+    if (env_var) {
+        sample_frequency = strtoul(env_var, NULL, 10);
+    }
+    env_var = memkind_get_env("PEBS_FREQ_HZ");
+    if (env_var) {
+        pebs_freq_hz = strtof(env_var, NULL);
```
Comment (maciekpac) [inline src/memkind_memtier.c:813]: Check for zero - if zero is supplied, segfault appears in pebs.c, line 82

Comment (maciekpac) [reply src/memkind_memtier.c:813]: check that value > 0; otherwise, calculations and units do not make sense

File: src/memkind_memtier.c [hunk 5]
```diff
@@ -98,6 +98,12 @@
 #define THRESHOLD_CHECK_CNT 20
 #define THRESHOLD_STEP      1024
 
+//PEBS
+    float old_time_window_hotness_weight;
+    float pebs_freq_hz;
+    unsigned long sample_frequency;
```
Comment (maciekpac) [inline src/memkind_memtier.c:104]: This should be a float/double - we might want to test lower frequencies, e.g. 0.5 Hz or 0.1 Hz as well.

Please remember about adding some basic checks

**Review by PatKamin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/3)* status: 2 of 4 files reviewed, 6 unresolved discussions (waiting on @bratpiorka, @maciekpac, and @PatKamin)

___
*[src/memkind_memtier.c, line 813 at r1](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/3#-MlUPuV-d1KaAryH4vEH-r1-813:-Mm2ZJV_2lANyv01tTpf:b-896fix) ([raw file](https://github.com/bratpiorka/memkind_hot_poc/blob/a8ccafb5bb0f676b70e1b7aab311c84dab11a722/src/memkind_memtier.c#L813)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

check that value > 0; otherwise, calculations and units do not make sense
</blockquote></details>

Done.

**Review by bratpiorka — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/3)* status: all files reviewed, 5 unresolved discussions (waiting on @maciekpac)

**Review by bratpiorka — CHANGES_REQUESTED** [review]
> Reviewed 3 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/3)* status: 3 of 4 files reviewed, 6 unresolved discussions (waiting on @bratpiorka, @maciekpac, and @PatKamin)

___
*[src/ranking.cpp, line 506 at r5](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/3#-MmN3eH720nQQKDFq6g0:-MmN3eH720nQQKDFq6g1:b-389tr1) ([raw file](https://github.com/bratpiorka/memkind_hot_poc/blob/61e1cf3366288f23f2d1a2ece68a0b577dcfbcba/src/ranking.cpp#L506)):*
> ```cpp
> //                                     const struct ttype *updated_values)
> // {
> //     ranking_remove_internal(ranking, entry_to_update);
> ```

why this is changed? is it a mistake?

**Review by PatKamin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/3)* status: all files reviewed, 5 unresolved discussions (waiting on @maciekpac)

___
*[src/ranking.cpp, line 506 at r5](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/3#-MmN3eH720nQQKDFq6g0:-MmN4I9-7cgbOXK3f_U3:b-1c0456) ([raw file](https://github.com/bratpiorka/memkind_hot_poc/blob/61e1cf3366288f23f2d1a2ece68a0b577dcfbcba/src/ranking.cpp#L506)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

why this is changed? is it a mistake?
</blockquote></details>

There is no change here against the master branch.

**Pull request merged**